lib/chef/cookbook_uploader.rb

```diff
@@ -139,12 +139,14 @@ def uploader_function_for(file, checksum, url, checksums_to_upload)
 
     def validate_cookbooks
       cookbooks.each do |cb|
-        syntax_checker = Chef::Cookbook::SyntaxCheck.for_cookbook(cb.name)
-        Chef::Log.info("Validating ruby files")
-        exit(1) unless syntax_checker.validate_ruby_files
-        Chef::Log.info("Validating templates")
-        exit(1) unless syntax_checker.validate_templates
-        Chef::Log.info("Syntax OK")
+        unless cb.nil?
```

Comment:
> So I'd suggest that the cookbook_version_loader needs to be changed to just throw on an empty cookbook or any cookbook that lacks a metadata.json or metadata.rb or which does not have a name or a version attribute:
> 
> https://github.com/chef/chef/blob/bddd4852791df30a78da74fd7e432e7efc87cf54/lib/chef/cookbook/cookbook_version_loader.rb#L177-L183
> 
> That code needs to get changed to be a lot more strict and less permissive.

@lamont-granquist does not upload the blank cookbook.

```$ bundle exec knife cookbook upload blank_dir -c ~/workspace/chef/chef-repo/.chef/knife.rb
WARN: Found a directory blank_dir in the cookbook path, but it contains no cookbook files. skipping.
Uploaded 0 cookbooks.
```

Second in bulk upload if we raise error it will break uploading of other cookbooks so what it is currently doing is skipping the blank cookbook (which does not have any files) to upload by giving warning message and uploading rest.

```
$ bundle exec knife cookbook upload --all -c ~/workspace/chef/chef-repo/.chef/knife.rb
WARN: Found a directory test in the cookbook path, but it contains no cookbook files. skipping.
WARN: Found a directory blank_dir in the cookbook path, but it contains no cookbook files. skipping.
Uploading audit          [5.0.0]
Uploading compat_resource [12.19.1]
Uploading mangled        [0.1.0]
Uploading starter        [0.0.0]
Uploading test_freeze    [0.1.3]
Uploading without_metadata [0.1.0]
Uploaded all cookbooks.
```

This PR is only to fix the nil class error which is coming while uploading bulk cookbooks with one of the cookbook as blank and single blank cookbook without any files.